src/LiveSplit.Core/Model/RunMetadata.cs

```diff
@@ -117,6 +117,49 @@ public IDictionary<Variable, VariableValue> VariableValues
             }
         }
 
+        public class CustomVariable
+        {
+            public string Value = null;
+            public bool IsPermanent = false;
+
+            public CustomVariable Permanent()
+            {
+                IsPermanent = true;
+                return this;
+            }
+
+            public void SetValue(string value)
+            {
+                Value = value;
+            }
+
+            public void ClearValue()
+            {
+                Value = null;
+            }
+        }
+
+        public IDictionary<string, CustomVariable> CustomVariables;
```

Comment:
What's the benefit of using the interface here?  
What's the reason for making it mutable?  
Do not expose fields publicly.

```suggestion
        public IReadOnlyDictionary<string, CustomVariable> CustomVariables { get; }
```

Or, if truly necessary:
```suggestion
        public IReadOnlyDictionary<string, CustomVariable> CustomVariables { get; private set; }
```